File: core/movie/lib/src/main/java/org/pcsoft/app/jmc/movie/ui/panes/MovieProjectPane.java
package org.pcsoft.app.jmc.movie.ui.panes;

import javafx.beans.property.ObjectProperty;
import javafx.scene.layout.BorderPane;
import org.pcsoft.framework.jfex.mvvm.Fxml;

import java.io.File;

public class MovieProjectPane extends BorderPane {

    private final MovieProjectPaneView viewController;
    private final MovieProjectPaneViewModel viewModel;

    public MovieProjectPane() {
        final var tuple = Fxml.from(MovieProjectPaneView.class).withRoot(this).load();
        viewController = tuple.getViewController();
        viewModel = tuple.getViewModel();
    }

    public MovieProjectPane(File file) {
        this();
        viewModel.setFile(file);
    }

    public File getFile() {
        return viewModel.getFile();
    }

    public ObjectProperty<File> fileProperty() {
        return viewModel.fileProperty();
    }

    public void setFile(File file) {
        viewModel.setFile(file);
    }
}

File: core/movie/lib/src/main/kotlin/org/pcsoft/app/jmc/movie/ui/panes/MovieProjectPane.kt
package org.pcsoft.app.jmc.movie.ui.panes

import javafx.scene.layout.BorderPane
import org.pcsoft.framework.jfex.mvvm.Fxml
import java.io.File

class MovieProjectPane(file: File) : BorderPane() {

    private val viewController: MovieProjectPaneView
    private val viewModel: MovieProjectPaneViewModel

    init {
        val tuple = Fxml.from(MovieProjectPaneView::class.java).withRoot(this).load()
        viewController = tuple.viewController
        viewModel = tuple.viewModel

        viewModel.file.set(file)
    }

    val file get() = viewModel.file
}